modules/carousel/jetpack-carousel.php

```diff
@@ -62,10 +66,14 @@ function init() {
 			 * @param array $this->prebuilt_widths Array of default widths.
 			 */
 			$this->prebuilt_widths = apply_filters( 'jp_carousel_widths', $this->prebuilt_widths );
-			add_filter( 'post_gallery', array( $this, 'enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
+			add_filter( 'post_gallery', array( $this, 'check_and_enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
 			add_filter( 'post_gallery', array( $this, 'set_in_gallery' ), -1000 );
 			add_filter( 'gallery_style', array( $this, 'add_data_to_container' ) );
 			add_filter( 'wp_get_attachment_image_attributes', array( $this, 'add_data_to_images' ), 10, 2 );
+			if ( $this->single_image_gallery_enabled ) {
+				add_filter( 'the_content', array( $this, 'add_data_to_single_images' ) );
+				$this->enqueue_assets();
```

Comment:
ATM it looks like this:
```
if( gallery shortcode is present ) {
  if( no_other_plugin_is_handling_gallery_shortcode ) {
    enqueue_assets()
  }
}
if( single_image_gallery_enabled ) {
  enqueue_assets()
}
```
So no - we're enqueuing the assets on all post pages, however in a bit different way.
BTW: condition for enqueuing the assets for pages without `[gallery]` will have to be modified to check if other plugins are not processing `[gallery]` shortcode (see my comment above).